### PR TITLE
RHPAM-3954 :Import a project via SSH is not working over a proxy

### PR DESCRIPTION
**Pull request by RishiRajAnand**
- this adds a new system property (**org.uberfire.nio.git.proxy.type**) to choose between 3 proxy types i.e http, sock5, sock4.
- For ex: to connect to a proxy server on sock5 protocol and with host 127.0.0.1 and port 8888 it looks like this. 
**-Dorg.uberfire.nio.git.proxy.ssh.over.http=true -Dhttp.proxyHost=127.0.0.1 -Dhttp.proxyPort=8888 -Dorg.uberfire.nio.git.proxy.type=sock5**
- Default  value of this property is "http" for backward compatibility

**Thank you for submitting this pull request**

**RHPAM-3954**: [_(please edit the JIRA link if it exists)_ ](https://issues.redhat.com/browse/RHPAM-3954)

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by RishiRajAnand** [comment]
Jenkins run fdb

**Comment by RishiRajAnand** [comment]
Jenkins run fdb

**Review by adrielparedes — CHANGES_REQUESTED** [review]

**Comment by RishiRajAnand** [comment]
Jenkins run fdb

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitSSHConfigSessionFactory.java [hunk 1]
```diff
@@ -69,10 +73,19 @@ ProxyHTTP buildProxy(final JGitFileSystemProviderConfiguration config) {
             user = config.getHttpsProxyUser();
             passw = config.getHttpsProxyPassword();
         }
-        final ProxyHTTP proxyHTTP = new ProxyHTTP(host, port);
-        if (user != null) {
+        if (config.getProxyType().equals("sock5")) {
```
[the remark this answers is not in the document]
Comment (RishiRajAnand) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitSSHConfigSessionFactory.java]: I have added the changes, Thanks

**Review by adrielparedes — APPROVED** [review]

**Review by tomasdavidorg — APPROVED** [review]
Thanks @RishiRajAnand . Tested and it looks good to me.

**Comment by RishiRajAnand** [comment]
FDB errors are unrelated

**Comment by akumar074** [comment]
@adrielparedes @tomasdavidorg @pefernan @RishiRajAnand 
Hey guys, is this PR good to merge? As this issue targets 7.12.1 we will need to raise backport PR as well.

**Comment by tomasdavidorg** [comment]
@akumar074 Looks good to me from my side.

**Comment by akumar074** [comment]
Thanks @tomasdavidorg 
@adrielparedes @pefernan Can you please merge this PR

**Pull request merged**